From 6686acc589e7bd35b56340407ac8316c406faf0b Mon Sep 17 00:00:00 2001
From: Dezhi Kong <dezhi.kong@amlogic.com>
Date: Mon, 3 Aug 2020 16:10:50 +0800
Subject: [PATCH 1/5] gpu: optimize gpu dvfs tbl [1/1]

PD#SWPL-30505

Problem:
the fps is lower when run westeros_test

Solution:
optimize the dvfs of gpu

Verify:
Local

Change-Id: I8fdc13ea505dac211e9ffac35d38630ae139ce42
Signed-off-by: Dezhi Kong <dezhi.kong@amlogic.com>
---
 arch/arm/boot/dts/amlogic/mesong12a_drm.dtsi   | 26 ++++++++++++++-------
 arch/arm/boot/dts/amlogic/mesong12b_drm.dtsi   | 32 ++++++++++++++++++++++++++
 arch/arm/boot/dts/amlogic/mesonsm1_drm.dtsi    | 32 ++++++++++++++++++++++++++
 arch/arm/boot/dts/amlogic/mesontl1_drm.dtsi    | 32 ++++++++++++++++++++++++++
 arch/arm/boot/dts/amlogic/mesontm2_drm.dtsi    | 32 ++++++++++++++++++++++++++
 arch/arm64/boot/dts/amlogic/mesong12a_drm.dtsi | 26 ++++++++++++++-------
 arch/arm64/boot/dts/amlogic/mesong12b_drm.dtsi | 32 ++++++++++++++++++++++++++
 arch/arm64/boot/dts/amlogic/mesonsm1_drm.dtsi  | 32 ++++++++++++++++++++++++++
 arch/arm64/boot/dts/amlogic/mesontl1_drm.dtsi  | 32 ++++++++++++++++++++++++++
 arch/arm64/boot/dts/amlogic/mesontm2_drm.dtsi  | 32 ++++++++++++++++++++++++++
 10 files changed, 292 insertions(+), 16 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/mesong12a_drm.dtsi b/arch/arm/boot/dts/amlogic/mesong12a_drm.dtsi
index 9d1e3f6..99436ac 100644
--- a/arch/arm/boot/dts/amlogic/mesong12a_drm.dtsi
+++ b/arch/arm/boot/dts/amlogic/mesong12a_drm.dtsi
@@ -283,23 +283,33 @@
 };
 
 &gpu{
-	/*gpu max freq is 850M*/
 	def_clk = <1>;
-	tbl = <&dvfs285_cfg &dvfs666_cfg &dvfs850_cfg &dvfs850_cfg>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs850_cfg
+		&dvfs850_cfg>;
 
 	dvfs285_cfg:dvfs285_cfg {
-		 keep_count = <2>;
-		 threshold = <100 200>;
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
 	};
 
 	dvfs666_cfg:dvfs666_cfg {
-		 keep_count = <1>;
-		 threshold = <85 200>;
+		 threshold = <80 120>;
 	};
 
 	dvfs850_cfg:dvfs850_cfg {
-		 keep_count = <1>;
-		 threshold = <179 255>;
+		 threshold = <80 255>;
 	};
 
 };
+
diff --git a/arch/arm/boot/dts/amlogic/mesong12b_drm.dtsi b/arch/arm/boot/dts/amlogic/mesong12b_drm.dtsi
index 7a01106..0e19ac1 100644
--- a/arch/arm/boot/dts/amlogic/mesong12b_drm.dtsi
+++ b/arch/arm/boot/dts/amlogic/mesong12b_drm.dtsi
@@ -280,3 +280,35 @@
 		};
 	};
 };
+
+&gpu{
+	def_clk = <1>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs800_cfg
+		&dvfs800_cfg>;
+
+	dvfs285_cfg:dvfs285_cfg {
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs666_cfg:dvfs666_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs800_cfg:dvfs800_cfg {
+		 threshold = <80 255>;
+	};
+
+};
+
diff --git a/arch/arm/boot/dts/amlogic/mesonsm1_drm.dtsi b/arch/arm/boot/dts/amlogic/mesonsm1_drm.dtsi
index 3ae18ba..9c19b88 100644
--- a/arch/arm/boot/dts/amlogic/mesonsm1_drm.dtsi
+++ b/arch/arm/boot/dts/amlogic/mesonsm1_drm.dtsi
@@ -281,3 +281,35 @@
 		};
 	};
 };
+
+&gpu{
+	def_clk = <1>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs850_cfg
+		&dvfs850_cfg>;
+
+	dvfs285_cfg:dvfs285_cfg {
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs666_cfg:dvfs666_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs850_cfg:dvfs850_cfg {
+		 threshold = <80 255>;
+	};
+
+};
+
diff --git a/arch/arm/boot/dts/amlogic/mesontl1_drm.dtsi b/arch/arm/boot/dts/amlogic/mesontl1_drm.dtsi
index 1060aad..0974b21 100644
--- a/arch/arm/boot/dts/amlogic/mesontl1_drm.dtsi
+++ b/arch/arm/boot/dts/amlogic/mesontl1_drm.dtsi
@@ -251,3 +251,35 @@
 		};
 	};
 };
+
+&gpu{
+	def_clk = <1>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs800_cfg
+		&dvfs800_cfg>;
+
+	dvfs285_cfg:dvfs285_cfg {
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs666_cfg:dvfs666_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs800_cfg:dvfs800_cfg {
+		 threshold = <80 255>;
+	};
+
+};
+
diff --git a/arch/arm/boot/dts/amlogic/mesontm2_drm.dtsi b/arch/arm/boot/dts/amlogic/mesontm2_drm.dtsi
index a08af5d..d8337e2 100644
--- a/arch/arm/boot/dts/amlogic/mesontm2_drm.dtsi
+++ b/arch/arm/boot/dts/amlogic/mesontm2_drm.dtsi
@@ -276,3 +276,35 @@
 		};
 	};
 };
+
+&gpu{
+	def_clk = <1>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs800_cfg
+		&dvfs800_cfg>;
+
+	dvfs285_cfg:dvfs285_cfg {
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs666_cfg:dvfs666_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs800_cfg:dvfs800_cfg {
+		 threshold = <80 255>;
+	};
+
+};
+
diff --git a/arch/arm64/boot/dts/amlogic/mesong12a_drm.dtsi b/arch/arm64/boot/dts/amlogic/mesong12a_drm.dtsi
index 06e1ad3..37e04fa 100644
--- a/arch/arm64/boot/dts/amlogic/mesong12a_drm.dtsi
+++ b/arch/arm64/boot/dts/amlogic/mesong12a_drm.dtsi
@@ -283,23 +283,33 @@
 };
 
 &gpu{
-	/*gpu max freq is 850M*/
 	def_clk = <1>;
-	tbl = <&dvfs285_cfg &dvfs666_cfg &dvfs850_cfg &dvfs850_cfg>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs850_cfg
+		&dvfs850_cfg>;
 
 	dvfs285_cfg:dvfs285_cfg {
-		 keep_count = <2>;
-		 threshold = <100 200>;
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
 	};
 
 	dvfs666_cfg:dvfs666_cfg {
-		 keep_count = <1>;
-		 threshold = <85 200>;
+		 threshold = <80 120>;
 	};
 
 	dvfs850_cfg:dvfs850_cfg {
-		 keep_count = <1>;
-		 threshold = <179 255>;
+		 threshold = <80 255>;
 	};
 
 };
+
diff --git a/arch/arm64/boot/dts/amlogic/mesong12b_drm.dtsi b/arch/arm64/boot/dts/amlogic/mesong12b_drm.dtsi
index fbdcb01..41ce4a7 100644
--- a/arch/arm64/boot/dts/amlogic/mesong12b_drm.dtsi
+++ b/arch/arm64/boot/dts/amlogic/mesong12b_drm.dtsi
@@ -280,3 +280,35 @@
 		};
 	};
 };
+
+&gpu{
+	def_clk = <1>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs800_cfg
+		&dvfs800_cfg>;
+
+	dvfs285_cfg:dvfs285_cfg {
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs666_cfg:dvfs666_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs800_cfg:dvfs800_cfg {
+		 threshold = <80 255>;
+	};
+
+};
+
diff --git a/arch/arm64/boot/dts/amlogic/mesonsm1_drm.dtsi b/arch/arm64/boot/dts/amlogic/mesonsm1_drm.dtsi
index 0da0378..43de163 100644
--- a/arch/arm64/boot/dts/amlogic/mesonsm1_drm.dtsi
+++ b/arch/arm64/boot/dts/amlogic/mesonsm1_drm.dtsi
@@ -281,3 +281,35 @@
 		};
 	};
 };
+
+&gpu{
+	def_clk = <1>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs850_cfg
+		&dvfs850_cfg>;
+
+	dvfs285_cfg:dvfs285_cfg {
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs666_cfg:dvfs666_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs850_cfg:dvfs850_cfg {
+		 threshold = <80 255>;
+	};
+
+};
+
diff --git a/arch/arm64/boot/dts/amlogic/mesontl1_drm.dtsi b/arch/arm64/boot/dts/amlogic/mesontl1_drm.dtsi
index 338f6c2..2ca57a4 100644
--- a/arch/arm64/boot/dts/amlogic/mesontl1_drm.dtsi
+++ b/arch/arm64/boot/dts/amlogic/mesontl1_drm.dtsi
@@ -251,3 +251,35 @@
 		};
 	};
 };
+
+&gpu{
+	def_clk = <1>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs800_cfg
+		&dvfs800_cfg>;
+
+	dvfs285_cfg:dvfs285_cfg {
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs666_cfg:dvfs666_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs800_cfg:dvfs800_cfg {
+		 threshold = <80 255>;
+	};
+
+};
+
diff --git a/arch/arm64/boot/dts/amlogic/mesontm2_drm.dtsi b/arch/arm64/boot/dts/amlogic/mesontm2_drm.dtsi
index 36ab12a..5e5a8de 100644
--- a/arch/arm64/boot/dts/amlogic/mesontm2_drm.dtsi
+++ b/arch/arm64/boot/dts/amlogic/mesontm2_drm.dtsi
@@ -281,3 +281,35 @@
 		};
 	};
 };
+
+&gpu{
+	def_clk = <1>;
+	tbl =  <&dvfs285_cfg
+		&dvfs400_cfg
+		&dvfs500_cfg
+		&dvfs666_cfg
+		&dvfs800_cfg
+		&dvfs800_cfg>;
+
+	dvfs285_cfg:dvfs285_cfg {
+		 threshold = <30 120>;
+	};
+
+	dvfs400_cfg:dvfs400_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs500_cfg:dvfs500_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs666_cfg:dvfs666_cfg {
+		 threshold = <80 120>;
+	};
+
+	dvfs800_cfg:dvfs800_cfg {
+		 threshold = <80 255>;
+	};
+
+};
+
-- 
2.7.4

